src/cpu/dyn_cache.h

```diff
@@ -783,15 +783,19 @@ static inline void dyn_mem_write(MAYBE_UNUSED void *ptr, MAYBE_UNUSED size_t siz
 
 static inline void dyn_cache_invalidate(MAYBE_UNUSED void *ptr, MAYBE_UNUSED size_t size)
 {
-#if defined(HAVE_SYS_ICACHE_INVALIDATE)
+#if defined(HAVE_BUILTIN_CLEAR_CACHE)
+	__builtin___clear_cache(static_cast<char *>(ptr),
+	                        reinterpret_cast<char *>(
+	                                reinterpret_cast<uintptr_t>(ptr) + size));
```

Comment:
Thanks for the reformat! Immediately easier to absorb the casts.
Did a quick test here to see if we could use pointer arithmetic - https://onlinegdb.com/-fGkNtQs6
From that, it looks like this should be do-able:
``` c++
	const auto addr = static_cast<char *>(ptr);
	__builtin___clear_cache(addr, addr + size);
```

Realize this is super minor.. but makes it appear less fragile/exotic?